File: src/main/interview/hourGlassMatrix/HourGlassMatrix.java
package main.interview.hourGlassMatrix;


import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class HourGlassMatrix {
    public static void main(String[] args) {
        List<Integer> sumList = new ArrayList<>();
        int rows = 6, columns = 6;

        int[][] array = new int[rows][columns];

        Scanner sc = new Scanner(System.in);

        for (int i = 0; i < array.length; i++) {
            for (int j = 0; j < array.length; j++) {
                array[i][j] = sc.nextInt();
            }
        }

        for(int i =0;i<4;i++) {
            for(int j=0;j<4;j++) {
               sumList.add(calculateSumOfHourGlass(array, i,j));
            }
        }
        System.out.println(Collections.max(sumList));
    }

    private static int calculateSumOfHourGlass(int[][] array, int i, int j) {
        return (array[i][j] + array[i][j+1] + array[i][j+2] + array[i+1][j+1] + array[i+2][j] + array[i+2][j+1] + array[i+2][j+2]);
    }
}

File: src/main/interview/hourGlassMatrix/HourglassMatrix.kt
package main.interview.hourGlassMatrix

import java.util.*

fun main() {
    val sumList = ArrayList<Int>()
    val rows = 6
    val columns = 6

    val array = Array(rows) {IntArray(columns)}

    val sc = Scanner(System.`in`)

    for (i in array.indices) {
        for (j in array.indices) {
            array[i][j] = sc.nextInt()
        }
    }

    for (i in 0..3) {
        for (j in 0..3) {
            sumList.add(calculateSumOfHourGlass(array, i, j))
        }
    }

    println(Collections.max(sumList))
}

private fun calculateSumOfHourGlass(array: Array<IntArray>, i: Int, j: Int): Int {
    return array[i][j] + array[i][j + 1] + array[i][j + 2] + array[i + 1][j + 1] + array[i + 2][j] + array[i + 2][j + 1] + array[i + 2][j + 2]
}